crates/re_viewer/src/ui/view_spatial/scene/picking.rs

```diff
@@ -207,14 +207,19 @@ pub fn picking(
     if state.closest_opaque_pick.instance_path_hash == InstancePathHash::NONE {
         if let Some(gpu_picking_result) = gpu_picking_result {
             // TODO(andreas): Pick middle pixel for now. But we soon want to snap to the closest object using a bigger picking rect.
-            let rect = gpu_picking_result.rect;
-            let picked_id = gpu_picking_result.picking_data
-                [(rect.width() / 2 + (rect.height() / 2) * rect.width()) as usize];
+            let picking_rect_position = gpu_picking_result.rect.extent / 2;
```

Comment:
I was very confused at first ~~
```suggestion
            let pos_on_picking_rect = gpu_picking_result.rect.extent / 2;